libsolutil/JSON.h

```diff
@@ -67,4 +67,67 @@ std::string jsonPrint(Json::Value const& _input, JsonFormat const& _format);
 /// \return \c true if the document was successfully parsed, \c false if an error occurred.
 bool jsonParseStrict(std::string const& _input, Json::Value& _json, std::string* _errs = nullptr);
 
+namespace detail
+{
+
+template<typename T>
+struct helper;
+
+#define DEFINE_HELPER(TYPE, CHECK_TYPE, CONVERT_TYPE)                                                     \
```

Comment:
maybe more specific name or undef macro

Reply:
undef

Reply:
Yes, you should definitely undef that. Macros are not affected by namespaces so this will be available globally wherever you include `JSON.h`. And I think we do include that in some prominent headers.